blacs/output_classes.py

```diff
@@ -54,22 +51,30 @@ def __init__(self, hardware_name, connection_name, device_name, program_function
         # Initialise Calibrations
         self._comboboxmodel.appendRow(QStandardItem(self._base_unit))
         if calib_class is not None:
-            if calib_class not in globals() or not isinstance(calib_params,dict) or globals()[calib_class].base_unit != default_units:
+            try:
+                cls = get_unit_conversion_class(calib_class)
+            # ImportError if module doesn't exist. Attribute error if it does but the
+            # class name does not exist within it. KeyError if the unit conversion class
+            # is an old-style unqualified class name expected to be in the globals()
+            # dict of the unitconversions module, but does not exist.
+            except (ImportError, AttributeError, KeyError):
+                cls = None
+            if cls is None or not isinstance(calib_params, dict) or cls.base_unit != default_units:
                 # log an error:  
                 reason = ''
-                if calib_class not in globals():
-                    reason = 'The unit conversion class was not imported. Is it in the correct folder? Is it imported when you call "from unitconversions import *" from a python terminal?'
-                elif not isinstance(calib_params,dict):
+                if calib_class is None:
+                    reason = f'The unit conversion class {calib_class} could not be imported. Ensure it is available on the computer running BLACS.'
+                elif not isinstance(calib_params, dict):
                     reason = 'The parameters for the unit conversion class are not a dictionary. Check your connection table code for errors and recompile it'
-                elif globals()[calib_class].base_unit != default_units:
+                elif cls.base_unit != default_units:
                     reason = 'The base unit of your unit conversion class does not match this hardware channel. The hardware channel has base units %s while your unit conversion class uses %s'%(globals()[calib_class].base_unit,default_units)
```

Comment:
This shouldn't use `globals()[calib_class]`

Reply:
I think..(not sure If I understand the code now)

Reply:
If it isn't clear from the GitHub highlight above, the error message still uses `globals()[calib_class]`

Reply:
Oh yes, that's wrong. Fixing.

Reply:
Fixed, but isn't this error message wrong? It prints the two units the wrong way around, attributing the class's base units to the hardware channel and vice versa.

Reply:
Yes you are right. It is the wrong way around

Reply:
Fixed